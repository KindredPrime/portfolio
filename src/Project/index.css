.Project {
  padding-bottom: 25px;
}

.Project__title {
  margin-top: 15px;
  margin-bottom: 5px;
}

.Project__description {
  margin-top: 15px;
}