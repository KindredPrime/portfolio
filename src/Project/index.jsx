import PropTypes from 'prop-types';
import TechLogos from '../TechLogos/index';
import ProjectLinks from '../ProjectLinks/index';
import './index.css';

function Project(props) {
  const {
    screenshots,
    title,
    techLogos,
    description,
    repoLink,
    liveLink
  } = props;

  return (
    <section className="Project">
      {screenshots.length > 0
        && <img src={screenshots[0].src} alt={screenshots[0].alt} />}
      <h2 className="Project__title">{title}</h2>
      <TechLogos logos={techLogos} />
      <p className="Project__description">{description}</p>
      <ProjectLinks repo={repoLink} live={liveLink} />
    </section>
  );
}

Project.propTypes = {
  screenshots: PropTypes.arrayOf(
    PropTypes.shape({
      src: PropTypes.string.isRequired,
      alt: PropTypes.string.isRequired
    })
  ).isRequired,
  title: PropTypes.string.isRequired,
  techLogos: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.number.isRequired,
      src: PropTypes.string.isRequired,
      alt: PropTypes.string.isRequired,
      classNames: PropTypes.string.isRequired
    }
  )),
  description: PropTypes.string.isRequired,
  repoLink: PropTypes.string.isRequired,
  liveLink: PropTypes.string.isRequired
};

export default Project;