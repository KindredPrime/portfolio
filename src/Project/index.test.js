import ReactDOM from 'react-dom';
import { render } from '@testing-library/react';
import Project from './index';

describe('Project Component', () => {
  it('renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(
      <Project
        screenshots={[]}
        title=""
        techLogos={[]}
        description=""
        repoLink=""
        liveLink=""
      />,
      div
    );
    ReactDOM.unmountComponentAtNode(div);
  });

  it('renders content as expected', () => {
    render(<Project
      screenshots={[]}
      title="Test Title"
      techLogos={[]}
      description="Test description"
      repoLink="https://www.repo.com"
      liveLink="https://www.live.com"
    />);

    expect(document.body).toMatchSnapshot();
  });
});