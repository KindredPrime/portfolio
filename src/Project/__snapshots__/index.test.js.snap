// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Project Component renders content as expected 1`] = `
<body>
  <div>
    <section
      class="Project"
    >
      <div
        class="ProjectSlideshow"
      />
      <h2
        class="Project__title"
      >
        Test Title
      </h2>
      <div
        class="TechLogos"
      />
      <p
        class="Project__description"
      >
        Test description
      </p>
      <div
        class="ProjectLinks"
      >
        <a
          class="ProjectLinks__repo"
          href="https://www.repo.com"
          rel="noreferrer"
          target="_blank"
        >
          GitHub
        </a>
        <span
          class="ProjectLinks__divider"
        >
          |
        </span>
        <a
          class="ProjectLinks__live"
          href="https://www.live.com"
          rel="noreferrer"
          target="_blank"
        >
          Live
        </a>
      </div>
    </section>
  </div>
</body>
`;
